compile error when using OpenBLAS in Ubuntu 22.04

environment:  
* ITensor version: `3.1.11`
* OS version: `Ubuntu 22.04`

`options.mk`  
```makefile
PLATFORM=openblas
BLAS_LAPACK_LIBFLAGS=-lpthread -L/usr/local/opt/openblas/lib -lopenblas
BLAS_LAPACK_INCLUDEFLAGS=-I/usr/local/opt/openblas/include -fpermissive -DHAVE_LAPACK_CONFIG_H -DLAPACK_COMPLEX_STRUCTURE
```

I got errors when using the `make` command in this repo. The following is the first error.
```
tensor/lapack_wrap.cc: In function ‘void itensor::dsyev_wrapper(char, char, itensor::LAPACK_INT, itensor::LAPACK_REAL*, itensor::LAPACK_REAL*, itensor::LAPACK_INT&)’:
tensor/lapack_wrap.cc:342:19: error: too few arguments to function ‘void dsyev_(const char*, const char*, const int*, double*, const int*, double*, double*, const int*, int*, size_t, size_t)’
  342 |     F77NAME(dsyev)(&jobz,&uplo,&n,A,&lda,eigs,&wkopt,&lwork,&info);
      |                   ^
In file included from /usr/include/lapack.h:11,
                 from /usr/include/lapacke.h:36,
                 from /home/zfb/itensor/itensor/tensor/lapack_wrap.h:50,
                 from tensor/lapack_wrap.cc:16:
/usr/include/lapack.h:17008:6: note: declared here
17008 | void LAPACK_dsyev_base(
      |      ^~~~~~~~~~~~~~~~~
tensor/lapack_wrap.cc:345:19: error: too few arguments to function ‘void dsyev_(const char*, const char*, const int*, double*, const int*, double*, double*, const int*, int*, size_t, size_t)’
  345 |     F77NAME(dsyev)(&jobz,&uplo,&n,A,&lda,eigs,work.data(),&lwork,&info);
      |                   ^
In file included from /usr/include/lapack.h:11,
                 from /usr/include/lapacke.h:36,
                 from /home/zfb/itensor/itensor/tensor/lapack_wrap.h:50,
                 from tensor/lapack_wrap.cc:16:
/usr/include/lapack.h:17008:6: note: declared here
17008 | void LAPACK_dsyev_base(
      |      ^~~~~~~~~~~~~~~~~
```

--- a/itensor/tensor/lapack_wrap.h
+++ b/itensor/tensor/lapack_wrap.h
@@ -149,7 +149,11 @@ imagRef(LAPACK_COMPLEX & z)
 #ifdef FORTRAN_NO_TRAILING_UNDERSCORE
 #define F77NAME(x) x
 #else
+#if defined(LAPACK_GLOBAL) || defined(LAPACK_NAME)
+#define F77NAME(x) LAPACK_##x
+#else
 #define F77NAME(x) x##_
+#endif
 #endif
 
 namespace itensor {

